### PR TITLE
custom github status

### PR DESCRIPTION
**Pull request by StarfallProjects**

### CODE REVIEW

File: pages/integrations/github.md.erb [hunk 1]
```diff
@@ -79,3 +79,42 @@ If you want to run builds *only* on pull requests, set the Branch Filter Pattern
...
 
 When you [connect your GitHub account to Buildkite](#connecting-buildkite-and-github) the email address associated with the GitHub account is added to your Buildkite account. If you've got GitHub set not to display your email, `[username]@users.noreply.github.com` or the more recent `[username+id]@users.noreply.github.com` is added instead. The email address of a commit is one of the ways Buildkite matches webhook builds to users.
+
+## Customizing commit statuses
+
+The commit status is the label used to identify the Buildkite checks on your commits and pull requests. Normally, Buildkite autogenerates these statuses. 
+
+For example: if you select _Update commit statuses_ in your _Pipeline Settings_, your checks will appear on your pull request as _buildkite/`<pipeline-name>`_.
+
+<%= image "update-commit-statuses-on-to-github.png", alt: "Screenshot of GitHub build settings and the resulting GitHub pull request statuses" %>
```
Comment (plaindocs) [inline pages/integrations/github.md.erb:89]: I find this image super hard to parse. What do you think about showing the setting image first, then "your checks will appear on your pull request as buildkite/<pipeline-name>" and then the image of the check?

File: pages/integrations/github.md.erb [hunk 2]
```diff
@@ -79,3 +79,42 @@ If you want to run builds *only* on pull requests, set the Branch Filter Pattern
...
+
+Sometimes, you may wish to customize these statuses. This can be especially useful if you reuse the same pipeline for multiple components in a monorepo.
+
+You can do this at both the build and step level, using the [`notify`](/docs/pipelines/notifications) attribute in your `pipeline.yml`.
```
Comment (plaindocs) [inline pages/integrations/github.md.erb:93]: What do you think about this?

```suggestion
You can customize the commit statuses, for example to reuse the same pipeline for multiple components in a monorepo,  at both the build and step level, using the [`notify`](/docs/pipelines/notifications) attribute in your `pipeline.yml`.
```

File: pages/integrations/github.md.erb [hunk 3]
```diff
@@ -79,3 +79,42 @@ If you want to run builds *only* on pull requests, set the Branch Filter Pattern
...
+
+### Build level
+
+1. Add the following to your `pipeline.yml`, not nested under any other parameters:
```
Comment (plaindocs) [inline pages/integrations/github.md.erb:97]: We probably need to add a section to the style guide about talking about YAML.

I'd say "At the top level of `pipeline.yml`. And if we are more specific, we've called it attributes above and elsewhere.

File: pages/integrations/github.md.erb [hunk 4]
```diff
@@ -79,3 +79,42 @@ If you want to run builds *only* on pull requests, set the Branch Filter Pattern
...
+
+    ```yaml
+    notify:
+    - github_commit_status:
+        context: "my-custom-status"
```
Comment (plaindocs) [inline pages/integrations/github.md.erb:102]:
```suggestion
    notify:
      - github_commit_status:
        context: "my-custom-status"
```

File: pages/integrations/github.md.erb [hunk 5]
```diff
@@ -79,3 +79,42 @@ If you want to run builds *only* on pull requests, set the Branch Filter Pattern
...
+
+For example: if you select _Update commit statuses_ in your _Pipeline Settings_, your checks will appear on your pull request as _buildkite/`<pipeline-name>`_.
+
+<%= image "update-commit-statuses-on-to-github.png", alt: "Screenshot of GitHub build settings and the resulting GitHub pull request statuses" %>
+
+Sometimes, you may wish to customize these statuses. This can be especially useful if you reuse the same pipeline for multiple components in a monorepo.
+
+You can do this at both the build and step level, using the [`notify`](/docs/pipelines/notifications) attribute in your `pipeline.yml`.
+
+### Build level
```
Comment (plaindocs) [inline pages/integrations/github.md.erb:95]: Would it make more sense to make these something like 

> Custom commit status for a pipeline
> Custom commit status for a command step?

File: pages/integrations/github.md.erb [hunk 6]
```diff
@@ -79,3 +79,42 @@ If you want to run builds *only* on pull requests, set the Branch Filter Pattern
...
+
+1. Add the following to your `pipeline.yml`, not nested under any other parameters:
+
+    ```yaml
+    notify:
+    - github_commit_status:
+        context: "my-custom-status"
+    ```
+<%= render_markdown 'integrations/step_2_3_github_custom_status' %>
```
Comment (plaindocs) [inline pages/integrations/github.md.erb:104]: The numbering breaks with this include (and the indentation also?), to be honest for type off stuff I'd just duplicate it.

File: styleguide/STYLE.md [hunk 7]
```diff
@@ -192,6 +192,20 @@ Internal links need to start with `/docs`, for example:
 Read more about [environment variables](/docs/pipelines/environment-variables)
 ```
 
+### Content reuse
+
+Add snippets to the directory where they'll be used, prefaced with an underscore. For example `_my_snippet.md.erb`. **However**, when pulling the snippet into a file, remove the leading underscore.
+
+Thus
```
Comment (plaindocs) [inline styleguide/STYLE.md:199]: can we avoid thus, even in the style guide?

**Review by plaindocs — CHANGES_REQUESTED** [review]
This is great, a short clear explanation. :+1: 

I've made quite a few suggestions at the cosmetic / wording level, let me know what you think

File: pages/integrations/github.md.erb (hunk 3 above)
Comment (StarfallProjects) [reply pages/integrations/github.md.erb:97]: Fixed and noted in style

File: pages/integrations/github.md.erb (hunk 1 above)
Comment (StarfallProjects) [reply pages/integrations/github.md.erb:89]: Edited

File: pages/integrations/github.md.erb (hunk 5 above)
Comment (StarfallProjects) [reply pages/integrations/github.md.erb:95]: I'm honestly not sure, but I don't think so. They are all custom statuses for the same pipeline?

File: pages/integrations/github.md.erb (hunk 6 above)
Comment (StarfallProjects) [reply pages/integrations/github.md.erb:104]: Done

File: styleguide/STYLE.md (hunk 7 above)
Comment (StarfallProjects) [reply styleguide/STYLE.md:199]: thus -> so

File: pages/integrations/github.md.erb (hunk 4 above)
Comment (StarfallProjects) [reply pages/integrations/github.md.erb:102]: Required a bit more tweaking (the context has to be further indented I think, per comment by Rose here: https://3.basecamp.com/3453178/buckets/1713315/messages/3455949494#__recording_3480020795 Have edited

**Comment by StarfallProjects** [comment]
Popping this back to draft status while I confirm some details of the more complex example

File: pages/integrations/github.md.erb [hunk 8]
```diff
@@ -79,3 +79,68 @@ If you want to run builds *only* on pull requests, set the Branch Filter Pattern
 ## Noreply email handling
 
 When you [connect your GitHub account to Buildkite](#connecting-buildkite-and-github) the email address associated with the GitHub account is added to your Buildkite account. If you've got GitHub set not to display your email, `[username]@users.noreply.github.com` or the more recent `[username+id]@users.noreply.github.com` is added instead. The email address of a commit is one of the ways Buildkite matches webhook builds to users.
+
+## Customizing commit statuses
+
+The commit status is the label used to identify the Buildkite checks on your commits and pull requests. Normally, Buildkite autogenerates these statuses.
```
Comment (plaindocs) [inline pages/integrations/github.md.erb:85]: Should we mention that this is on GitHub anywhere?

File: pages/integrations/github.md.erb [hunk 9]
```diff
@@ -79,3 +79,68 @@ If you want to run builds *only* on pull requests, set the Branch Filter Pattern
...
+For example, if you select _Update commit statuses_ in your _Pipeline Settings_:
+
+<%= image "update-commit-statuses-on.png", alt: "Screenshot of GitHub build settings with Update commit statuses enabled" %>
+
+Your checks will appear on your pull request as _buildkite/`<pipeline-name>`_:
+
+<%= image "github-default-status.png", alt: "Screenshot of the resulting GitHub pull request statuses" %>
+
+
+You can customize the commit statuses, for example to reuse the same pipeline for multiple components in a monorepo,  at both the build and step level, using the [`notify`](/docs/pipelines/notifications) attribute in your `pipeline.yml`.
```
Comment (plaindocs) [inline pages/integrations/github.md.erb:96]: :+1:

Comment (plaindocs) [inline pages/integrations/github.md.erb:96]:
```suggestion
You can customize the commit statuses, for example to reuse the same pipeline for multiple components in a monorepo, at both the build and step level, using the [`notify`](/docs/pipelines/notifications) attribute in your `pipeline.yml`.
```

Comment (plaindocs) [reply pages/integrations/github.md.erb:96]: So maybe the approach is something like 

> You can customize the commit statuses using the [`notify`](/docs/pipelines/notifications) attribute, either for the entire pipeline or just for one or more steps, for example to reuse the same pipeline for multiple components in a monorepo.

File: pages/integrations/github.md.erb [hunk 10]
```diff
@@ -79,3 +79,68 @@ If you want to run builds *only* on pull requests, set the Branch Filter Pattern
...
+    ```yaml
+    notify:
+        - github_commit_status:
+            context: "my-custom-status"
+    ```
+2. In _Pipeline Settings_ > _GitHub_, make sure _Update commit statuses_ is not selected. Note that this prevents Buildkite from automatically creating and sending statuses for this pipeline, meaning you will have to handle all commit statuses through the `pipeline.yml`.
+3. When you make a new commit or pull request, you should see _my-custom-status_ as the commit status:
+    <%= image "github-custom-status.png", alt: "Screenshot of GitHub build settings and the resulting GitHub pull request statuses" %>
+
+In a basic setup (one `pipeline.yml` for a repo containing one codebase), this customizes the commit status for the pipeline. However, if you have multiple `pipeline.yml` files in one repo, feeding in to the same Buildkite pipeline, this allows you to have different statuses when building different sections of the repo.
```
Comment (plaindocs) [inline pages/integrations/github.md.erb:111]: Thoughts on 

```suggestion
In a setup for a repository containing one codebase and one `pipeline.yml`, this customizes the commit status for the pipeline. However, if you have multiple `pipeline.yml` files in one repo, feeding in to the same Buildkite pipeline, this allows you to have different statuses when building different sections of the repo.
```

File: pages/integrations/github.md.erb [hunk 11]
```diff
@@ -79,3 +79,68 @@ If you want to run builds *only* on pull requests, set the Branch Filter Pattern
...
+
+For example, if you have a monorepo containing three applications, you could use the same pipeline, with different `pipeline.yml` files for each application:
```
Comment (plaindocs) [inline pages/integrations/github.md.erb:113]: This breaks my brain. :-D

File: pages/integrations/github.md.erb [hunk 12]
```diff
@@ -79,3 +79,68 @@ If you want to run builds *only* on pull requests, set the Branch Filter Pattern
...
+
+```
```
Comment (plaindocs) [inline pages/integrations/github.md.erb:115]: Would a screenshot of actual files makes sense?

**Review by plaindocs — APPROVED** [review]
Looks good to me. 
Address or ignore any of these comments as you like

File: pages/integrations/github.md.erb (hunk 8 above)
Comment (StarfallProjects) [reply pages/integrations/github.md.erb:85]: It's in the GitHub docs? But sure, edited

File: pages/integrations/github.md.erb (hunk 12 above)
Comment (StarfallProjects) [reply pages/integrations/github.md.erb:115]: Ah this was a made up example. I can create some files and screenshot them on GitHub if you wish? Although I wouldn't be able to get all the directories and individual yamls into one screenshot . . .

File: pages/integrations/github.md.erb (hunk 11 above)
Comment (StarfallProjects) [reply pages/integrations/github.md.erb:113]: Because I've worded it unclearly or because it's a complex setup? I have to say it seems a WEIRD setup - I did some digging for monorepos, and most people using them seem to dynamically create pipelines, rather than trying to reuse one for everything. And I'd be reassured if someone from the original thread gave these docs a look :p But yeah . .  .

File: pages/integrations/github.md.erb (hunk 12 above)
Comment (plaindocs) [reply pages/integrations/github.md.erb:115]: You could literally do it on the local file system for this example? Show the same 3 dirs + 3 files but not in plain text?

Comment (plaindocs) [reply pages/integrations/github.md.erb:115]: I've seen a thing that converts these diagrams to images, but can't remember what it was.

File: pages/integrations/github.md.erb (hunk 11 above)
Comment (plaindocs) [reply pages/integrations/github.md.erb:113]: Apologies again. Just because it isn't how my brain expects it.

File: pages/integrations/github.md.erb (hunk 12 above)
Comment (StarfallProjects) [reply pages/integrations/github.md.erb:115]: Am I being fabulously daft? How do you simultaneously screenshot three same-level dirs and the files inside them?
